### PR TITLE
CI: Add a workflow to automatically label pull requests

### PR DESCRIPTION
**Pull request by dfandrich**
The labeler language is quite restrictive right now so labels are added
quite conservatively, meaning that many PRs won't get labels when it's
"obvious" they should. It will still save some manual work on those
that it can label.

If this labeler were already running, it would have labeled this PR with "CI".

### CODE REVIEW

**Comment by dfandrich** [comment]
Actually, the action *is* running and it *did* add a CI label!

File: .github/labeler.yml [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+- all: ['docs/WebSockets.md']
+- all: ['docs/libcurl/opts/CURLOPT_WS_OPTIONS.3']
+- all: ['docs/libcurl/curl_ws_*']
+
+Windows:
+- all: ['projects/Windows/**']
+- all: ['lib/*win32*']
+- all: ['lib/curl_multibyte.*']
+- all: ['lib/rename.*']
+- all: ['Winbuild/**']
```
Comment (jay) [inline .github/labeler.yml:164]: If this is case sensitive then it is 'winbuild' in lowercase. Also I think it should be added to the build tag if the labeler allows multiple tags to be applied to the same path.

Comment (dfandrich) [reply .github/labeler.yml]: Done and done.

**Comment by dfandrich** [comment]
I've added a few more documentation files & build paths.

**Review by jay — APPROVED** [review]
Let's see what happens.

**Comment by dfandrich** [comment]
The first workflow runs ran with *GITHUB_TOKEN Permissions  PullRequests: write* but now they're running with *PullRequests: read* and failing with the error `Error: HttpError: Resource not accessible by integration`. The only difference that seems relevant is that the original PRs were on the curl repo but the PRs were on other people's repos. I wonder if that is the difference.

**Comment by dfandrich** [comment]
I think that's it. According to [the docs](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token), *Maximum access for
pull requests from public forked repositories: pull-requests=read* so GitHub is downgrading the write access we want on other people's PRs. That makes this workflow almost useless.

**Comment by jay** [comment]
what about s/pull_request/pull_request_target like [this guy](https://github.com/actions/labeler/issues/399)

**Comment by dfandrich** [comment]
I didn't notice that `pull_request_target` difference; I used the default template that was spit out for this action that had `pull_request`. I've submitted that change that, hopefully, will improve things. ReviewPad is considerably more powerful and can automate more things for us, but if people are hesitant about adding another app to the curl repo, and the labeler action merges my PR to add *allofany* patterns, this workflow will handle 90% of the labels I'd like to create ATM.

**Pull request closed without merging**